Added a bunch of deriving Eq

geekingfrog opened the pull request:
As mentioned in #72 I added a bunch of `deriving Eq` for most of the types exported by the library.

jtdaugherty commented:
Thanks for the patch!  Some of the instances you added make sense for simpler types like `ViewportType`, but I'm not sure about others.  For example, I think that if you're ever in a situation where you need to compare `AttrMap` values for equality, there's something else that could be done instead.  And you should never be comparing `Context` values for equality because there is only one.  In general adding instances implies that certain usage patterns are anticipated, and I wan't to discourage usage patterns that rely on using equality with some data structures.  Were those instances added out of a need for those specific operations?  Or were they added out of a desire for completeness?

Here are the instances I think are worth keeping in the patch:

* `VisibilityRequest` `Eq`
* `ViewportType` `Eq`
* `Direction` `Show`, `Eq`
* `Location` `Eq`
* `CursorLocation` `Eq` (although here, too, it's unlikely that you'll ever need equality on these)
* `BorderStyle` `Eq`

geekingfrog commented:
You're right about implying usage through the types. I added some more for the sake of consistency than any real need, but you have a good point.

I kept the `Eq` instances for the types you mentionned above and removed the other (namely, `AttrMap`, `Viewport` and `Context`). (ff325d978679cddb18609d968de7885c65f3abb8)

Thanks for the feedback.

jtdaugherty commented:
Okay. Thank you!
